src/plugins/intel_gpu/src/graph/impls/ocl/primitive_base.hpp

```diff
@@ -207,6 +205,18 @@ struct typed_primitive_impl_ocl : public typed_primitive_impl<PType> {
         }
     }
 
+    void set_arguments_impl(typed_primitive_inst<PType>& instance, kernel_arguments_data& args) override {
+        if (instance.can_be_optimized() || is_cpu()) {
```

Comment:
This `is_cpu()` check seems to be useless, as classes derived from `typed_primitive_impl_ocl` are always non-cpu impls